src/rez/utils/filesystem.py

```diff
@@ -697,3 +698,22 @@ def windows_long_path(dos_path):
         path = "\\\\?\\" + path
 
     return path
+
+
+def compare_location(loc1, loc2):
```

Comment:
This function is kind of opinionated. Both paths are lower cased but only the first path is normalized (through canonical_path). Also the exception mentions "repository".

So to have a function in src/rez/utils/filesystem.py is alright, but the function would need to be generic and non opinionated. ALso I'd name it `compare_paths`.

Additionally, a unit test would be welcome.